components/layout_2020/replaced.rs

```diff
@@ -82,25 +118,83 @@ impl ReplacedContent {
             .max_box_size()
             .percentages_relative_to(containing_block);
 
-        let clamp = |inline_size: Length, block_size: Length| {
-            (
-                inline_size.clamp_between_extremums(min_box_size.inline, max_box_size.inline),
-                block_size.clamp_between_extremums(min_box_size.block, max_box_size.block),
-            )
+        let default_object_size = || {
+            // FIXME:
+            // “If 300px is too wide to fit the device, UAs should use the width of
+            //  the largest rectangle that has a 2:1 ratio and fits the device instead.”
+            // “height of the largest rectangle that has a 2:1 ratio, has a height not greater
+            //  than 150px, and has a width not greater than the device width.”
+            physical::Vec2 {
+                x: Length::new(300.),
+                y: Length::new(150.),
+            }
+            .size_to_flow_relative(mode)
+        };
+        let clamp = |inline_size: Length, block_size: Length| Vec2 {
+            inline: inline_size.clamp_between_extremums(min_box_size.inline, max_box_size.inline),
+            block: block_size.clamp_between_extremums(min_box_size.block, max_box_size.block),
         };
         // https://drafts.csswg.org/css2/visudet.html#min-max-widths
         // https://drafts.csswg.org/css2/visudet.html#min-max-heights
-        let (inline_size, block_size) = match (box_size.inline, box_size.block) {
+        match (box_size.inline, box_size.block) {
```

Comment:
Returning `Vec2<_>` values made the code more verbose, IMO, but it's not very important.

Reply:
For some reason I thought this change helped having an early return, but it looks like it wasn’t necessary after all.